content/class/Poll.md

```diff
@@ -0,0 +1,117 @@
+| ------ | --------- | -------- |
+| Object | [Poll](https://developer.twitter.com/en/docs/twitter-api/data-dictionary/object-model/poll) | [Poll](https://docs.joinmastodon.org/entities/poll/) |
+
+## Properties
+
+### `id` {#property_id}
+
+![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public string id { get; }
+```
+
+The unique identifier for this Poll.
+
+| API   | Twitter 2 | Mastodon |
+| ----- | --------- | -------- |
+| Field | `id`      | `id`     |
+| Type  | string    | string   |
+
+### `end_date` {#property_end_date}
+
+![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public GLib.Date end_date { get; }
+```
+
+The date when the poll ends.
+
+| API   | Twitter 2       | Mastodon        |
+| ----- | --------------- | --------------- |
+| Field | `end_datetime`  | `expires_at`    |
+| Type  | ISO 8601 string | ISO 8601 string |
+
+### `options` {#property_options}
+
+![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public PollOption[] options { get; }
+```
+
+All options this Poll contains. See [`PollOoption`](../structure/PollOption.md).
+
+| API   | Twitter 2 | Mastodon  |
+| ----- | --------- | --------- |
+| Field | `options` | `options` |
+| Type  | array     | array     |
+
+## Creation Methods
+
+### `Poll.from_json` {#creation_method_from_json}
+
+![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public Poll.from_json (Json.Object obj)
+```
+
+Creates a Poll Object from a given Json object.
+
+## Methods
+
+### `has_flag` {#method_has_flag}
+
+![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public bool has_flag (PollFlag flag)
+```
+
+Returns if the Poll has a specific flag set.
+
+### `set_flag` {#method_set_flag}
+
+![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public void set_flag (PollFlag flag)
+```
+
+Sets a flag for this Poll.
+
+### `unset_flag` {#method_unset_flag}
+
+![API: Twitter 2](https://img.shields.io/badge/API-Twitter%202-blue?style=flat-square) ![API: Mastodon](https://img.shields.io/badge/API-Mastodon-purple?style=flat-square)
+
+```c#
+public void unset_flag (PollFlag flag)
+```
+
+Removes a flag from this Poll.
+
+## Fields
+
+### `flags` {#field_flags}
+
+```c#
+private Gee.HashSet<PollFlag> flags
```

Comment:
If these are proper enum values, can we not just bitmask them together rather than dragging in libgee and having a hashset of them?

Reply:
The HashSet was the closest I could find as being similar to a bitmask in Vala (I mean, it also don't have dictionaries to my knowledge)...

Reply:
Vala supports bitmasking of flags without any additional collections: https://github.com/IBBoard/cawbird/blob/master/src/list/TweetListEntry.vala#L306

Dictionaries are called [HashMaps](https://valadoc.org/gee-0.8/Gee.HashMap.html) in libgee or [HashTable](https://valadoc.org/glib-2.0/GLib.HashTable.html) in plain glibc.

Reply:
Alright, didn't know of this (but to my defense I could not find it in the documentation...).

Will look into this.

Reply:
Yeah, it's probably tricky to search for.

And according to the docs that I've just found then there's a better way to do it in pure Vala!

https://wiki.gnome.org/Projects/Vala/Manual/Enumerated%20types%20(Enums)#Flag_types

Our other code is referencing a C enum that's then imported via the vaapi file, and each value is defined by bit-shifting.